extensions/standard-processors/tests/unit/FetchFileTests.cpp

```diff
@@ -0,0 +1,348 @@
+  REQUIRE(file_contents.size() == 1);
+  REQUIRE(file_contents[0].empty());
+  using org::apache::nifi::minifi::utils::verifyLogLinePresenceInPollTime;
+  REQUIRE(verifyLogLinePresenceInPollTime(1s, "[warning] Read permission denied for file"));
+}
+#endif
+
+TEST_CASE_METHOD(FetchFileTestFixture, "Test fetching file with default file path", "[testFetchFile]") {
+  test_controller_.runSession(plan_);
+  auto file_contents = getSuccessfulFlowFileContents();
+  REQUIRE(file_contents.size() == 1);
+  REQUIRE(file_contents[0] == file_content_);
+  REQUIRE(utils::file::FileUtils::exists(input_dir_ + utils::file::FileUtils::get_separator() + input_file_name_));
+}
+
+TEST_CASE_METHOD(FetchFileTestFixture, "Test fetching file from a custom path", "[testFetchFile]") {
+  REQUIRE(0 == utils::file::FileUtils::create_dir(input_dir_ + utils::file::FileUtils::get_separator() + "sub"));
+  utils::putFileToDir(input_dir_ + utils::file::FileUtils::get_separator() + "sub", input_file_name_, file_content_);
+  auto file_path = input_dir_ + utils::file::FileUtils::get_separator() + "sub" + utils::file::FileUtils::get_separator() + input_file_name_;
+  plan_->setProperty(fetch_file_processor_, org::apache::nifi::minifi::processors::FetchFile::FileToFetch.getName(), file_path);
+  test_controller_.runSession(plan_);
+  auto file_contents = getSuccessfulFlowFileContents();
+  REQUIRE(file_contents.size() == 1);
+  REQUIRE(file_contents[0] == file_content_);
+  REQUIRE(utils::file::FileUtils::exists(file_path));
+}
+
+TEST_CASE_METHOD(FetchFileTestFixture, "Flow scheduling fails due to missing move destination directory when completion strategy is set to move file", "[testFetchFile]") {
+  plan_->setProperty(fetch_file_processor_, org::apache::nifi::minifi::processors::FetchFile::CompletionStrategy.getName(), "Move File");
+  REQUIRE_THROWS_AS(test_controller_.runSession(plan_), minifi::Exception);
+}
+
+TEST_CASE_METHOD(FetchFileTestFixture, "Flow fails due to move conflict", "[testFetchFile]") {
+  auto move_dir = test_controller_.createTempDirectory();
+  utils::putFileToDir(move_dir, input_file_name_, "old content");
+  plan_->setProperty(fetch_file_processor_, org::apache::nifi::minifi::processors::FetchFile::CompletionStrategy.getName(), "Move File");
+  plan_->setProperty(fetch_file_processor_, org::apache::nifi::minifi::processors::FetchFile::MoveDestinationDirectory.getName(), move_dir);
+  plan_->setProperty(fetch_file_processor_, org::apache::nifi::minifi::processors::FetchFile::MoveConflictStrategy.getName(), "Fail");
+  test_controller_.runSession(plan_);
+  auto file_contents = getFailedFlowFileContents();
+  REQUIRE(file_contents.size() == 1);
+  REQUIRE(file_contents[0].empty());
+
+  std::ifstream is(move_dir + utils::file::FileUtils::get_separator() + input_file_name_, std::ifstream::binary);
+  REQUIRE(std::string((std::istreambuf_iterator<char>(is)), std::istreambuf_iterator<char>()) == "old content");
+}
```

Comment:
we could also check that `input_dir_/input_file_name_` is still there

Reply:
Added check in 5cc8e67a59aa66ed854bdb3349b241c549357c4a